pkg/controller/istiocontrolplane/listeners.go

```diff
@@ -90,7 +90,7 @@ func (u *IstioStatusUpdater) EndReconcile(_ runtime.Object, status map[string]*v
 		return fmt.Errorf("failed to get IstioOperator before updating status due to %v", err)
 	}
 	iop.Spec.ComponentStatus = status
```

Comment:
I think this line here is the issue. An operator should never update the `spec` of its input CRs

Reply:
Reason being that you will run into conflicts much more often if you update the spec. In general, `spec` should only be updated by the user, `status` only by the operator. This way, conflicts are much less likely

Reply:
See my comments: https://github.com/istio/istio/issues/20021#issuecomment-572466483

Reply:
The PR https://github.com/istio/operator/pull/713 has move status to spec, need to get the reason that change.

Reply:
Yes- sorry I didn't realize the new CRD was a complete rewrite. Will investigate this a bit. I think we should move the status out of the spec again, but that will now require a PR to istio/api

Reply:
> The PR #713 moved status to spec. Moving status to spec is not a good idea, but I need to get the reasons for that change.

@ostromart @howardjohn Can we get a change in istio/api to get the status filed outside spec?

Reply:
Done https://github.com/istio/api/pull/1231